tests/test_connection.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+from trio_websocket import ConnectionClosed, WebSocketClient, WebSocketServer
+import trio
+
+
+import logging
+logging.basicConfig(level=logging.DEBUG)
+HOST = 'localhost'
+RESOURCE = 'resource'
+
+
+@pytest.fixture
+async def echo_server(nursery):
+    async def handler(conn):
+        try:
+            msg = await conn.get_message()
+            await conn.send_message(msg)
+        except ConnectionClosed:
+            pass
+    server = WebSocketServer(handler, HOST, 0, ssl_context=None)
+    await nursery.start(server.listen)
+    yield server
+
+
+def client_for_server(server):
+    ''' Create a client configured to connect to ``server``. '''
+    return WebSocketClient(HOST, server.port, RESOURCE, use_ssl=False)
+
+
+async def test_client_send_and_receive(echo_server, nursery):
+    client = client_for_server(echo_server)
+    async with await client.connect(nursery) as conn:
```

Comment:
I'm still finding it very odd to be passing in a nursery in this API

please see e.g. asyncwebsockets which is a native trio library with reasonable API and shutdown behavior, and doesn't pass nurseries around

Reply:
This is the most important design decision that needs to be made at this moment. Let's discuss it over here: https://github.com/HyperionGray/trio-websocket/issues/20